endpoint_info.go

```diff
@@ -159,7 +159,11 @@ func (ep *endpoint) Info() EndpointInfo {
 		return ep
 	}
 
-	return sb.getEndpoint(ep.ID())
+	if epi := sb.getEndpoint(ep.ID()); epi != nil {
+		return epi
+	}
```

Comment:
this is well explained here : http://stackoverflow.com/questions/29138591/hiding-nil-values-understanding-why-golang-fails-here and I do see this as a problem.
I would suggest lets clean up other places in the code as well instead of fixing just this place.